ambry-network/src/main/java/com.github.ambry.network/BlockingChannel.java

```diff
@@ -38,8 +39,8 @@
   protected boolean connected = false;
   protected InputStream readChannel = null;
   protected WritableByteChannel writeChannel = null;
-  protected Object lock = new Object();
-  private SocketChannel channel = null;
+  protected final Object lock = new Object();
```

Comment:
Since there's only one lock, you can remove it and synchronize the methods that use it.

Reply:
I was trying to avoid making too many changes at once, but yes, I'm aware of locking on 'this' :) I can make the change, if everyone is OK with that